### PR TITLE
fix default profile where unsupported

### LINKED ISSUE #18946: seccomp: config provided but seccomp not supported.
Building from current master returns the following error any time any container is started:

```
# docker run --rm busybox true
seccomp: config provided but seccomp not supported
docker: Error response from daemon: Cannot start container 43f6e217197d8a286cf38bfef0361e2aafc5251d15da5f5850ea9c63caad7f12: [9] System error: seccomp: config provided but seccomp not supported.
```

daemon logs:
```
WARN[0005] exit status 1                                
WARN[0005] failed to cleanup ipc mounts:
failed to umount /var/lib/docker/containers/43f6e217197d8a286cf38bfef0361e2aafc5251d15da5f5850ea9c63caad7f12/shm: invalid argument
failed to umount /var/lib/docker/containers/43f6e217197d8a286cf38bfef0361e2aafc5251d15da5f5850ea9c63caad7f12/mqueue: invalid argument 
ERRO[0005] Error unmounting container 43f6e217197d8a286cf38bfef0361e2aafc5251d15da5f5850ea9c63caad7f12: not mounted 
ERRO[0005] Handler for POST /v1.22/containers/43f6e217197d8a286cf38bfef0361e2aafc5251d15da5f5850ea9c63caad7f12/start returned error: Cannot start container 43f6e217197d8a286cf38bfef0361e2aafc5251d15da5f5850ea9c63caad7f12: [9] System error: seccomp: config provided but seccomp not supported 
```

The package was built without the `seccomp` build tag. However having it as a build tag implies it is optional, and thus docker should work without it.

```
# docker version
Client:
 Version:      1.10.0-dev
 API version:  1.22
 Go version:   go1.5.1
 Git commit:   78ce43b
 Built:        Mon Dec 28 22:05:30 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.0-dev
 API version:  1.22
 Go version:   go1.5.1
 Git commit:   78ce43b
 Built:        Mon Dec 28 22:05:30 2015
 OS/Arch:      linux/amd64

# docker info                 
Containers: 4
Images: 55
Server Version: 1.10.0-dev
Storage Driver: btrfs
Execution Driver: native-0.2
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: null host bridge
Kernel Version: 4.0.5-gentoo
Operating System: Gentoo/Linux
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.64 GiB
Name: whistler
ID: 27PU:7FU5:JFWG:62CV:LVZ7:GKDH:5DEC:5I27:QDHP:MNWM:QBY6:BKGR
Username: phemmer
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support

```

Built from 78ce43bad85fdf97d7c836315e46b531901c8ae7

### PR DESCRIPTION
**Pull request by jessfraz**
closes #18946 ping @phemmer

### CODE REVIEW

**Comment by phemmer** [comment]
Build fails:

```
vendor/src/github.com/docker/docker/daemon/execdriver/native/seccomp_unsupported.go:5: use of untyped nil
```

**Comment by phemmer** [comment]
There we go, all good now (with d161413).

**Comment by jfrazelle** [comment]
ya just testing on a server to spare you :)

File: daemon/execdriver/native/seccomp_unsupported.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// +build linux,!seccomp
+
+package native
+
+import "github.com/opencontainers/runc/libcontainer/configs"
+
+var (
+	seccompSupported      = false
+	defaultSeccompProfile = &configs.Seccomp{}
```
Comment (crosbymichael) [inline daemon/execdriver/native/seccomp_unsupported.go]: could this just be `nil` and then you could probably remove the need for the boolean.  Anyways, nice job on this ;)

Comment (jfrazelle) [inline daemon/execdriver/native/seccomp_unsupported.go]: ya it doesnt like nil, that was my first choice tho then got the type error
:(

Comment (jfrazelle) [inline daemon/execdriver/native/seccomp_unsupported.go]: oh wow my b I know

**Comment by jfrazelle** [comment]
ok now its pretty, apparently i left my brain on vacation :)

**Comment by phemmer** [comment]
Still working with 94b4531.

Also, should the dependency be mentioned in https://github.com/docker/docker/blob/master/project/PACKAGERS.md#build-dependencies? Even though it's optional?

**Comment by jfrazelle** [comment]
good catch, I will add in #18949 commit, since it has to do with packaging

**Comment by calavera** [comment]
LGTM

**Comment by icecrime** [comment]
LGTM

**Pull request merged**